### PR TITLE
git-node: resturcture using commandDir of yargs

### PR DESCRIPTION
**Pull request by joyeecheung**
This uses the `commandDir` option of yargs to restructure the `git-node` commands.

Removes the arg tests because it's basically testing the functionality of yargs.

https://github.com/yargs/yargs/blob/HEAD/docs/advanced.md#commanddirdirectory-opts

### CODE REVIEW

**Comment by targos** [comment]
```
/home/travis/build/nodejs/node-core-utils/components/git/land.js
  1:13  error  Missing semicolon  semi
```

**Comment by joyeecheung** [comment]
Fixed linter and added examples in the cli. @targos PTAL, thanks!

**Comment by targos** [comment]
Can you post here how the examples look like in the console?

File: docs/git-node.md [hunk 1]
```diff
@@ -17,20 +19,101 @@ A custom Git command for managing pull requests. You can run it as
...
+
+Options:
+  --version  Show version number                                       [boolean]
+  --help     Show help                                                 [boolean]
+```
+
+### `git node land`
+
+```
+git-node land [prid|options]
```
Comment (joyeecheung) [inline docs/git-node.md]: Here

File: docs/git-node.md [hunk 2]
```diff
@@ -17,20 +19,101 @@ A custom Git command for managing pull requests. You can run it as
...
+  git node land --abort     Abort the current session
+  git node land --amend     Append metadata to the current commit message
+  git node land --final     Verify the landed PR and clean up
+  git node land --continue  Continue the current landing session
+```
+
+### `git node metadata`
+
+```
+git-node metadata <identifier>
```
Comment (joyeecheung) [inline docs/git-node.md]: And here

**Comment by targos** [comment]
What is the purpose of `git node land --continue`?

**Comment by joyeecheung** [comment]
@targos Basically when you forgot where you are at during a landing session, you can run this and it tries to continue the session for you...usually just tries to `--amend` if you are in the middle of a rebase and `--final` if you have done rebasing.

**Pull request merged**